### PR TITLE
Adds tweet search suggestion feature

### PR DESCRIPTION
**Pull request by SKrPl**
Solves issue: https://github.com/fossasia/loklak_wok_android/issues/27

Changes build.gradle files for increasing targetSdk and minSdk, updating
the gradle build tool version and adding support material design,
retrofit, gson and butterknife library.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,13 +1,13 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.2"
+    compileSdkVersion 25
+    buildToolsVersion '25.0.0'
```
Comment (abishekvashok) [inline app/build.gradle]: 25.0.3 is the latest, can you update to it?

File: app/build.gradle [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
     defaultConfig {
         applicationId "org.loklak.android.wok"
-        minSdkVersion 11
-        targetSdkVersion 23
+        minSdkVersion 14
```
Comment (abishekvashok) [inline app/build.gradle]: Is there any compatibility problem for SDK 11?

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -10,17 +10,26 @@
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
-        android:theme="@style/Theme.AppCompat.NoActionBar">
-        <activity android:name="org.loklak.android.wok.MainActivity">
+        android:theme="@style/AppTheme">
+        <!--<activity android:name=".MainActivity">
```
Comment (abishekvashok) [inline app/src/main/AndroidManifest.xml]: Better to remove code than to comment it. (Comment while you test :))

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -10,17 +10,26 @@
...
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
     </application>
 
-</manifest>
+</manifest>
```
Comment (abishekvashok) [inline app/src/main/AndroidManifest.xml]: Don't remove a newline at eof

File: app/src/main/java/org/loklak/android/adapters/SuggestAdapter.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    class SuggestViewHolder extends RecyclerView.ViewHolder {
+
+        @BindView(R.id.suggest_query)
+        TextView suggestQuery;
+
+        SuggestViewHolder(View itemView) {
+            super(itemView);
+            ButterKnife.bind(this, itemView);
+            // suggestQuery = (TextView) itemView.findViewById(R.id.suggest_query);
```
Comment (abishekvashok) [inline app/src/main/java/org/loklak/android/adapters/SuggestAdapter.java]: Follow up

File: app/src/main/java/org/loklak/android/model/suggest/Query.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    private Integer mTimezoneOffset;
+    private String mRetrievalNext;
+    private Integer mScoreRetrieval;
+    private String mQueryLast;
+    private String mExpectedNext;
+    private Integer mScoreSuggest;
+    private Integer mRetrievalCount;
+    private String mQueryFirst;
+
+    public Query() {
```
Comment (abishekvashok) [inline app/src/main/java/org/loklak/android/model/suggest/Query.java]: Empty constructor

File: app/src/main/java/org/loklak/android/model/suggest/SearchMetadata.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+public class SearchMetadata {
+
+    private String mCount;
+    private Integer mHits;
+    private String mQuery;
+    private String mOrder;
+    private String mOrderby;
+    private String mClient;
+
+    public SearchMetadata() {
```
Comment (abishekvashok) [inline app/src/main/java/org/loklak/android/model/suggest/SearchMetadata.java]: Empty constructor

File: app/src/main/res/layout/row_tweet_search_suggest.xml [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        android:padding="@dimen/suggest_query_padding"
+        android:gravity="center_vertical"
+        android:textSize="@dimen/suggest_query_font_size"/>
+
+    <View
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/divider_height"
+        android:background="@color/divider_color"/>
+
+</LinearLayout>
```
Comment (abishekvashok) [inline app/src/main/res/layout/row_tweet_search_suggest.xml]: Add a newline at eof

**Review by abishekvashok — CHANGES_REQUESTED** [review]
There are other empty constructors as well. See comments, the rest is fine :)

**Comment by abishekvashok** [comment]
Also as you changed build tools and target SDK update them in `.travis.yml` @SkrPl? That's why Travis is falling.
Also please reference #30

File: app/build.gradle (hunk 2 above)
Comment (SKrPl) [reply app/build.gradle]: Yes, material design library requires minSdk to be 14.

File: app/src/main/java/org/loklak/android/model/suggest/Query.java (hunk 6 above)
Comment (SKrPl) [reply app/src/main/java/org/loklak/android/model/suggest/Query.java]: I will be using RealmDatabase, which requires a constructor to be there.

File: app/build.gradle (hunk 1 above)
Comment (SKrPl) [reply app/build.gradle]: Sure :+1:

File: app/build.gradle (hunk 2 above)
Comment (abishekvashok) [reply app/build.gradle]: Ok

Comment (abishekvashok) [reply app/build.gradle]: But I read somewhere the up to 11 material design library is backward compatibile

File: app/src/main/java/org/loklak/android/model/suggest/Query.java (hunk 6 above)
Comment (abishekvashok) [reply app/src/main/java/org/loklak/android/model/suggest/Query.java]: Then can you use the constructor, by putting the initialization of the variables to be done inside the constructor

File: app/build.gradle (hunk 2 above)
Comment (SKrPl) [reply app/build.gradle]: Maybe due to incomplete build it was asking to use minSdk to be 14, but I again retried and it works. Will change it.

Comment (abishekvashok) [reply app/build.gradle]: Ok

**Comment by SKrPl** [comment]
@abishekvashok rebased it and have added other 4 functionalities mentioned in the issue.
@Orbiter @sudheesh001 @djmgit Please review it.

**Review by djmgit — COMMENTED** [review]
@SKrPl lokks good, would be nice if you could add a app acreen shot.

**Comment by SKrPl** [comment]
device-2017-06-07-121030

**Comment by djmgit** [comment]
@SKrPl looks nice, please check the travis failure

**Comment by abishekvashok** [comment]
Yea pls check travis

**Comment by SKrPl** [comment]
@abishekvashok  @djmgit rebased it, solved some of the lint issues.

File: app/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -10,17 +10,24 @@
...
 
-        <service
-            android:name=".HarvestService"
-            android:exported="false" />
+        <activity android:name="org.loklak.android.ui.activity.SearchActivity"></activity>
+
     </application>
 
-</manifest>
+</manifest>
```
Comment (abishekvashok) [inline app/src/main/AndroidManifest.xml]: Add a newline at eof

**Comment by SKrPl** [comment]
@abishekvashok Added newline at eof in manifest.xml

**Review by abishekvashok — APPROVED** [review]
It works

**Comment by mariobehling** [comment]
@SKrPl Please work more closely together with other developers to get your PRs reviewed and provide a scrum that let's others follow up more easily.

**Pull request merged**